### PR TITLE
New Plugin - Node Labels Resources

### PR DESCRIPTION
**Pull request by Sharathmk99**
New Liqo Resource Offering plugin to calculate resources based on node labels. This plugin is really useful when you want to dedicated some nodes to liqo and exclude nodes related to system or others.

Thanks

### CODE REVIEW

**Comment by Sharathmk99** [comment]
@cheina97 Please review this PR. Thanks

File: pkg/node-labels-resources/grpc_server.go [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+	}
+
+	nodeFactory := informers.NewSharedInformerFactoryWithOptions(
+		clientset, resyncPeriod, informers.WithTweakListOptions(noVirtualNodesFilter),
+	)
+	nodeInformer := nodeFactory.Core().V1().Nodes().Informer()
+	podFactory := informers.NewSharedInformerFactoryWithOptions(
+		clientset, resyncPeriod, informers.WithTweakListOptions(noShadowPodsFilter),
+	)
+	clientset.CoreV1().Nodes()
```
Comment (aleoli) [inline pkg/node-labels-resources/grpc_server.go:89]: Is this required?

File: pkg/node-labels-resources/grpc_server.go [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+	podInformer := podFactory.Core().V1().Pods().Informer()
+	s := NodeLabelsMonitor{
+		Server:        grpc.NewServer(),
+		nodeLabels:    nodeLabels,
+		allocatable:   corev1.ResourceList{},
+		k8sNodeClient: clientset.CoreV1().Nodes(),
+		k8sPodClient:  clientset.CoreV1().Pods(""),
```
Comment (aleoli) [inline pkg/node-labels-resources/grpc_server.go:96]:
```suggestion
		k8sPodClient:  clientset.CoreV1().Pods(corev1.NamespaceAll),
```

File: pkg/node-labels-resources/grpc_server.go [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+	if err != nil {
+		klog.V(1).Infof("error: error during sending notification to liqo: %s", err)
+	}
+
+	for {
+		<-ctx.Done()
+		nlm.subscribers.Delete(req)
+		klog.V(1).Infof("info: liqo controller manager disconnected")
+		return nil
+	}
```
Comment (aleoli) [inline pkg/node-labels-resources/grpc_server.go:291]:
```suggestion
    <-ctx.Done()
    nlm.subscribers.Delete(req)
    klog.V(1).Infof("info: liqo controller manager disconnected")
    return nil
```

The for loop is not required here

**Review by aleoli — COMMENTED** [review]
Hi @Sharathmk99!

It looks good to me. Please fix these small comments and rebase the pr; thanks!

File: cmd/node-labels-resources/doc.go [hunk 4]
```diff
@@ -0,0 +1 @@
+package main
```
Comment (aleoli) [inline cmd/node-labels-resources/doc.go:1]:
```suggestion
// package main contains the main function for the node-label-resource-plugin.
package main
```

Here you can fix te linter with something like this

File: pkg/node-labels-resources/doc.go [hunk 5]
```diff
@@ -0,0 +1 @@
+package nodeselectorresources
```
Comment (aleoli) [inline pkg/node-labels-resources/doc.go:1]:
```suggestion
// package nodeselectorresources contains the logic for the node-label-resource-plugin.
package nodeselectorresources
```

File: pkg/node-labels-resources/grpc_server.go [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+	"k8s.io/klog/v2"
+	resourcehelper "k8s.io/kubectl/pkg/util/resource"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/liqotech/liqo/pkg/consts"
+	resourcemonitors "github.com/liqotech/liqo/pkg/liqo-controller-manager/resource-request-controller/resource-monitors"
+	"github.com/liqotech/liqo/pkg/utils"
+)
+
+type NodeDetails struct {
```
Comment (aleoli) [inline pkg/node-labels-resources/grpc_server.go:33]: All exported types, fields, and functions need to have a comment.

File: pkg/node-labels-resources/grpc_server.go [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+	s := NodeLabelsMonitor{
+		Server:        grpc.NewServer(),
+		nodeLabels:    nodeLabels,
+		allocatable:   corev1.ResourceList{},
+		k8sNodeClient: clientset.CoreV1().Nodes(),
+		k8sPodClient:  clientset.CoreV1().Pods(corev1.NamespaceAll),
+		ctx:           ctx,
+		resourceLists: map[string]NodeDetails{},
+	}
+	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```
Comment (aleoli) [inline pkg/node-labels-resources/grpc_server.go:100]:
```suggestion
	utilruntime.Must(nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

The same for the other occurrence

**Review by aleoli — APPROVED** [review]

**Pull request merged**